contracts/DataFeeds/TokenPriceDataFeed.sol

```diff
@@ -15,16 +15,18 @@ contract TokenPriceDataFeed is DataFeedOracleBase {
   )
     public
   {
+    require(_exchangeAdapter.tokenPairExists(_token1, _token2), 'token pair does not exist');
+
     token1 = _token1;
     token2 = _token2;
     exchangeAdapter = _exchangeAdapter;
 
     // set dataSource to `this` so setResult can only be called from this contract
-    DataFeedOracleBase.initialize(address(this));
+    initialize(address(this));
```

Comment:
I feel like uuur the one always wanting to be specifc when calling extended classes